devicetypes/smartthings/testing/simulated-window-shade.src/simulated-window-shade.groovy

```diff
@@ -0,0 +1,183 @@
+/**
+ *  Copyright 2018 SmartThings
+ *
+ *  Provides a simulated window shade.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "Simulated Window Shade", namespace: "smartthings", author: "SmartThings", runLocally: false, mnmn: "SmartThings", vid: "generic-window-shade") {
+		capability "Actuator"
+		capability "Window Shade"
+		capability "Window Shade Preset"
+
+		// Commands to use in the simulator
+		command "openPartially"
+		command "closePartially"
+		command "partiallyOpen"
+		command "opening"
+		command "closing"
+		command "opened"
+		command "closed"
+		command "unknown"
+	}
+
+	preferences {}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"windowShade", type: "generic", width: 6, height: 4){
+			tileAttribute ("device.windowShade", key: "PRIMARY_CONTROL") {
+				attributeState "open", label:'${name}', action:"close", icon:"st.shades.shade-open", backgroundColor:"#79b821", nextState:"closing"
+				attributeState "closed", label:'${name}', action:"open", icon:"st.shades.shade-closed", backgroundColor:"#ffffff", nextState:"opening"
+				attributeState "partially open", label:'Open', action:"close", icon:"st.shades.shade-open", backgroundColor:"#79b821", nextState:"closing"
+				attributeState "opening", label:'${name}', action:"pause", icon:"st.shades.shade-opening", backgroundColor:"#79b821", nextState:"partially open"
+				attributeState "closing", label:'${name}', action:"pause", icon:"st.shades.shade-closing", backgroundColor:"#ffffff", nextState:"partially open"
+				attributeState "unknown", label:'${name}', action:"open", icon:"st.shades.shade-closing", backgroundColor:"#ffffff", nextState:"opening"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
```

Comment:
Are we adding `level` to Window Shade?

Reply:
It will be a separate capability. But that reminds me to add Switch Level back in.